builder/softlayer/client.go

```diff
@@ -182,9 +183,41 @@ func (self SoftlayerClient) doHttpRequest(path string, requestType string, reque
 		return nil, err
 	}
 
+	if err := self.hasErrors(decodedResponse); err != nil {
+		return nil, err
+	}
+
 	return decodedResponse, nil
 }
 
+func (self SoftlayerClient) doHttpRequestMany(path string, requestType string, requestBody *bytes.Buffer) ([]interface{}, error) {
+	responseBody, err := self.doRawHttpRequest(path, requestType, requestBody)
+	if err != nil {
+		err := errors.New(fmt.Sprintf("Failed to get proper HTTP response from SoftLayer API"))
+		return nil, err
+	}
+
+	var decodedResponse interface{}
+	err = json.Unmarshal(responseBody, &decodedResponse)
+	if err != nil {
+		err := errors.New(fmt.Sprintf("Failed to decode JSON response from SoftLayer: %s | %s", responseBody, err))
+		return nil, err
+	}
+
+	switch v := decodedResponse.(type) {
+	case []interface{}:
+		return v, nil
+	case map[string]interface{}:
+		if err := self.hasErrors(v); err != nil {
+			return nil, err
+		}
+
+		return nil, errors.New("Expected multiple results")
+	default:
+		return nil, errors.New("Expected multiple results")
+	}
+}
+
```

Comment:
Good points.

I thought about another possible approach. How about always returning an **Array**? 

In case the result is a **map** you can always wrap it with an Array (with one cell).
This way you will have one `doHttpRequest` function which will always return an array and you will need to do the type casting only once.

The downside of that approach is that you will need to change some additional code inside `client.go` so that other functions that expect `doHttpRequest` to return a map will use **Array** instead.